src/accelerate/accelerator.py

```diff
@@ -556,6 +556,16 @@ def main_process_first(self):
         Lets the main process go first inside a with block.
 
         The other processes will enter the with block after the main process exits.
+
+        Example:
+
+        ```python
+        >>> from accelerate import Accelerator
+
+        >>> accelerator = Accelerator()
+        >>> with accelerator.main_process_first():
+        ...     print(f"This will be printed by process {accelerator.process_index}")
```

Comment:
Maybe put something like "This will only be printed on process 0."?

Reply:
Since it's `main_process_first` and `local_process_first`, it will start with 0 and go up. I added a comment in the example stating something similar.

Reply:
It's actually a bit more complicated: it will be printed on process 0 first, then in a (seemingly) random order by the others.

Reply:
Good point, will update stating something similar